docs/tutorials/agent_framework/RAG_with_conversational_flow_agent.md

```diff
@@ -0,0 +1,395 @@
+# Topic
+
+> Agent Framework is an experimental feature released in OpenSearch 2.12 and not recommended for use in a production environment. For updates on the progress of the feature or if you want to leave feedback, see the associated [GitHub issue](https://github.com/opensearch-project/ml-commons/issues/1161).
+
+> This tutorial doesn't explain what's retrieval-augmented generation(RAG).
+
+This tutorial explains how to use conversational flow agent to build RAG application by leveraging your 
+OpenSearch data as knowledge base.
+
+Note: You should replace the placeholders with prefix `your_` with your own value
+
+# Steps
+
+## 0. Preparation
+
+To build RAG application, we need to have some OpenSearch index as knowledge base. In this tutorial, we
+are going to use [k-NN index](https://opensearch.org/docs/latest/search-plugins/knn/knn-index/) and 
+[semantic search](https://opensearch.org/docs/latest/search-plugins/semantic-search/). You can read more 
+details on their document and this [tutorial](https://opensearch.org/docs/latest/search-plugins/neural-search-tutorial/).
+It's totally fine to just follow below steps to quick start.
+
+### update cluster setting
+
+If you have dedicated ML node, you don't need to set `"plugins.ml_commons.only_run_on_ml_node": false`.
+
+We set `"plugins.ml_commons.native_memory_threshold"` as 100% to avoid triggering native memory circuit breaker.
+```
+PUT _cluster/settings
+{
+    "persistent": {
+        "plugins.ml_commons.only_run_on_ml_node": false,
+        "plugins.ml_commons.native_memory_threshold": 100,
+        "plugins.ml_commons.agent_framework_enabled": true,
+        "plugins.ml_commons.memory_feature_enabled": true
```

Comment:
isn't this enabled by default now?

Reply:
(`memory_feature_enabled` - didn't mean to highlight _all_ that)

Reply:
Checkd code, it's still disabled, @Zhangxunmt can you enable these once all security issues fixed ? We can remove this part later when it's enabled by default.